plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServiceAccountUtil.java

```diff
@@ -38,18 +38,18 @@ public class ServiceAccountUtil {
    * @param projectId GCP project ID for {@code serviceAccountId} 
    * @param destination path of a key file to be saved
    */
-  public static void createAppEngineDefaultServiceAccountKey(IGoogleApiFactory apiFactory,
-      Credential credential, String projectId, Path destination)
+  public static void createAppEngineDefaultServiceAccountKey(IGoogleApiFactory apiFactory, 
+      String projectId, Path destination)
           throws FileAlreadyExistsException, IOException {
-    Preconditions.checkNotNull(credential, "credential not given");
+    Preconditions.checkState(apiFactory.hasCredentialsSet(), "credentials not set");
```

Comment:
`Preconditions.checkState` --> "Ensures the truth of an expression involving the state of the calling instance, _but not involving any parameters to the calling method_."

`Preconditions.checkArgument` --> "Ensures the truth of an expression _involving one or more parameters to the calling method_."

From https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html